EventFiltering/PWGHF/HFFilter.cxx

```diff
@@ -235,6 +244,48 @@ struct HfFilter { // Main struct for HF triggers
       LOG(fatal) << "Can't apply optimisation if ML is not applied.";
     }
   }
+
+  /// compute TPC postcalibrated nsigma based on calibration histograms from CCDB
+  /// \param objectMean calibration histograms of mean from CCDB
+  /// \param objectSigma calibration histograms of sigma from CCDB
+  /// \param track is the track
+  /// \param chargePar is the PID species
+  /// \return the corrected Nsigma value for the PID species
+  template <typename P, typename T>
+  double GetPostTPCCalib(const P& objectMean, const P& objectSigma, const T& track, const int chargePar)
+  {
+    TH3F* calibMean = objectMean;
+    TH3F* calibSigma = objectSigma;
+
+    double kTPCncls = track.tpcNClsFound();
+    double kTPCPin = track.tpcInnerParam();
+    double kEta = track.eta();
+    double kTPCnSigma = 0.;
+
+    if (chargePar == kKa)
+      kTPCnSigma = track.tpcNSigmaKa();
+    if (chargePar == kPi)
+      kTPCnSigma = track.tpcNSigmaPi();
+    if (chargePar == kPr)
+      kTPCnSigma = track.tpcNSigmaPr();
+
+    int binTPCncls = calibMean->GetXaxis()->FindBin(kTPCncls);
+    binTPCncls = (binTPCncls == 0 ? 1 : binTPCncls);
+    binTPCncls = (binTPCncls > calibMean->GetXaxis()->GetNbins() ? calibMean->GetXaxis()->GetNbins() : binTPCncls);
+    int binPin = calibMean->GetYaxis()->FindBin(kTPCPin);
+    binPin = (binPin == 0 ? 1 : binPin);
+    binPin = (binPin > calibMean->GetYaxis()->GetNbins() ? calibMean->GetYaxis()->GetNbins() : binPin);
+    int binEta = calibMean->GetZaxis()->FindBin(kEta);
+    binEta = (binEta == 0 ? 1 : binEta);
+    binEta = (binEta > calibMean->GetZaxis()->GetNbins() ? calibMean->GetZaxis()->GetNbins() : binEta);
```

Comment:
You can use `std::min`, `std::max` instead of the conditions.

Reply:
I would keep it here for the moment.

Reply:
I mean
```c++
(binTPCncls > calibMean->GetXaxis()->GetNbins() ? calibMean->GetXaxis()->GetNbins() : binTPCncls)
```
is equivalent to 
```c++
std::min(calibMean->GetXaxis()->GetNbins(), binTPCncls)
```
which is much easier to read and less prone to errors.